### PR TITLE
Add a lot of script metadata

### PR DESCRIPTION
**Pull request by LordMonoxide**

### CODE REVIEW

File: src/main/java/legend/game/SMap.java [hunk 1]
```diff
@@ -1892,13 +2077,19 @@ public static FlowControl FUN_800e02fc(final RunningScript<?> script) {
     return FlowControl.CONTINUE;
   }
 
+  @ScriptDescription("Get the number of model parts in a submap object model")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "scriptIndex", description = "The SubmapObject210 script index")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "numberOfParts", description = "The number of model parts")
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/SMap.java:2082]: 1 should be OUT

File: src/main/java/legend/game/SMap.java [hunk 2]
```diff
@@ -3864,7 +4053,7 @@ public static long FUN_800e5534(final int newCut, final int newScene) {
     _800bd7b4.setu(0);
     if(_800cab28.get() == 0) {
       if(fullScreenEffect_800bb140._24 == 0) {
-        scriptStartEffect(1, 10);
+        Scus94491BpeSegment.startFadeEffect(1, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/SMap.java:4056]: Fix static import

File: src/main/java/legend/game/Scus94491BpeSegment.java [hunk 3]
```diff
@@ -2527,7 +2546,7 @@ public static int getSubmapMusicChange() {
       SubmapMusic08 a2;
       int a3;
       for(a3 = 0, a2 = _8004fb00.get(a3); a2.submapCut_00.get() != 99 || a2.musicIndex_02.get() != 99; a3++, a2 = _8004fb00.get(a3)) { // I think 99 is just a sentinel value that means "end of list"
-        final int submapIndex = submapIndex_800bd808.get();
+        final int submapIndex = submapId_800bd808.get();
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/Scus94491BpeSegment.java:2549]: Rename int to submapId for consistency?

File: src/main/java/legend/game/Scus94491BpeSegment_8002.java [hunk 4]
```diff
@@ -1088,7 +1087,7 @@ public static void loadAndRenderMenus() {
         deallocateRenderables(0xff);
         uiFile_800bdc3c = null;
 
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/Scus94491BpeSegment_8002.java:1090]: Fix static import

File: src/main/java/legend/game/combat/Bttl_800c.java [hunk 5]
```diff
@@ -1052,7 +1052,7 @@ public static void battleInitiateAndPreload_800c772c() {
 
     battleLoaded_800bc94c.set(true);
 
-    scriptStartEffect(4, 30);
+    Scus94491BpeSegment.startFadeEffect(4, 30);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/combat/Bttl_800c.java:1055]: Fix static import

File: src/main/java/legend/game/inventory/screens/AdditionsScreen.java [hunk 6]
```diff
@@ -54,7 +54,7 @@ protected void render() {
         this.charSlot = 0;
         this.selectedSlot = 0;
         this.additionHighlight = null;
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/AdditionsScreen.java:57]: Fix static import

File: src/main/java/legend/game/inventory/screens/CampaignSelectionScreen.java [hunk 7]
```diff
@@ -37,7 +37,7 @@
 public class CampaignSelectionScreen extends MenuScreen {
   public CampaignSelectionScreen() {
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/CampaignSelectionScreen.java:40]: Fix static import

File: src/main/java/legend/game/inventory/screens/CharSwapScreen.java [hunk 8]
```diff
@@ -42,7 +42,7 @@ public CharSwapScreen(final Runnable unload) {
   protected void render() {
     switch(this.loadingStage) {
       case 0 -> {
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/CharSwapScreen.java:45]: Fix static import

File: src/main/java/legend/game/inventory/screens/DabasScreen.java [hunk 9]
```diff
@@ -82,7 +82,7 @@ protected void render() {
         //TODO this is the Pocketstation minigame to upload to the memcard
         loadDrgnFile(0, 6668, file -> this.dabasFilePtr_8011dd00 = file);
 
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/DabasScreen.java:85]: Fix static import

File: src/main/java/legend/game/inventory/screens/EquipmentScreen.java [hunk 10]
```diff
@@ -61,7 +61,7 @@ public EquipmentScreen(final Runnable unload) {
   protected void render() {
     switch(this.loadingStage) {
       case 0:
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/EquipmentScreen.java:64]: Fix static import

File: src/main/java/legend/game/inventory/screens/GoodsScreen.java [hunk 11]
```diff
@@ -27,7 +27,7 @@ public class GoodsScreen extends MenuScreen {
 
   public GoodsScreen(final Runnable unload) {
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/GoodsScreen.java:30]: Fix static import

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java [hunk 12]
```diff
@@ -35,7 +35,7 @@ public class ItemListScreen extends MenuScreen {
 
   public ItemListScreen(final Runnable unload) {
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/ItemListScreen.java:38]: Fix static import

File: src/main/java/legend/game/inventory/screens/KeybindsScreen.java [hunk 13]
```diff
@@ -76,7 +76,7 @@ public KeybindsScreen(final ConfigCollection config, final Runnable unload) {
     this.addKey(GLFW_KEY_END, "END");
 
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/KeybindsScreen.java:79]: Fix static import

File: src/main/java/legend/game/inventory/screens/LoadGameScreen.java [hunk 14]
```diff
@@ -28,7 +28,7 @@ public LoadGameScreen(final Consumer<SavedGame> saveSelected, final Runnable clo
     this.closed = closed;
 
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/LoadGameScreen.java:31]: Fix static import

File: src/main/java/legend/game/inventory/screens/MainMenuScreen.java [hunk 15]
```diff
@@ -165,7 +165,7 @@ protected void render() {
     switch(this.loadingStage) {
       case 0 -> {
         cacheCharacterSlots();
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/MainMenuScreen.java:168]: Fix static import

File: src/main/java/legend/game/inventory/screens/ModsScreen.java [hunk 16]
```diff
@@ -22,7 +22,7 @@ public class ModsScreen extends VerticalLayoutScreen {
 
   public ModsScreen(final Set<String> enabledMods, final Runnable unload) {
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/ModsScreen.java:25]: Fix static import

File: src/main/java/legend/game/inventory/screens/NewCampaignScreen.java [hunk 17]
```diff
@@ -46,7 +46,7 @@ public NewCampaignScreen() {
     this.enabledMods.addAll(MODS.getAllModIds());
 
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/NewCampaignScreen.java:49]: Fix static import

File: src/main/java/legend/game/inventory/screens/OptionsScreen.java [hunk 18]
```diff
@@ -22,7 +22,7 @@ public class OptionsScreen extends VerticalLayoutScreen {
 
   public OptionsScreen(final ConfigCollection config, final Set<ConfigStorageLocation> validLocations, final Runnable unload) {
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/OptionsScreen.java:25]: Fix static import

File: src/main/java/legend/game/inventory/screens/SaveGameScreen.java [hunk 19]
```diff
@@ -35,7 +35,7 @@ public SaveGameScreen(final Runnable unload) {
     this.unload = unload;
 
     deallocateRenderables(0xff);
-    scriptStartEffect(2, 10);
+    Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/SaveGameScreen.java:38]: Fix static import

File: src/main/java/legend/game/inventory/screens/ShopScreen.java [hunk 20]
```diff
@@ -128,7 +128,7 @@ protected void render() {
 
       case AWAIT_INIT_1 -> {
         if(uiFile_800bdc3c != null) {
-          scriptStartEffect(2, 10);
+          Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/ShopScreen.java:131]: Fix static import

File: src/main/java/legend/game/inventory/screens/TooManyItemsScreen.java [hunk 21]
```diff
@@ -92,7 +92,7 @@ protected void render() {
       case _3 -> {
         deallocateRenderables(0);
         this.FUN_8010fd80(true, this.droppedItems.get(this.dropIndex).itemId_00, this.invIndex, this.invScroll, 0);
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/TooManyItemsScreen.java:95]: Fix static import

File: src/main/java/legend/game/inventory/screens/UseItemScreen.java [hunk 22]
```diff
@@ -83,7 +83,7 @@ public UseItemScreen(final Runnable unload) {
   protected void render() {
     switch(this.loadingStage) {
       case 0 -> {
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/UseItemScreen.java:86]: Fix static import

File: src/main/java/legend/game/modding/coremod/config/ControllerKeybindsConfigEntry.java [hunk 23]
```diff
@@ -1,23 +1,22 @@
...
-
 public class ControllerKeybindsConfigEntry extends ConfigEntry<Void> {
   public ControllerKeybindsConfigEntry() {
     super(null, ConfigStorageLocation.GLOBAL, o -> new byte[0], bytes -> null);
 
     this.setEditControl((current, configCollection) -> {
       final Button button = new Button(I18n.translate(CoreMod.MOD_ID + ".config.controller_keybinds.configure"));
       button.onPressed(() -> button.getScreen().getStack().pushScreen(new KeybindsScreen(configCollection, () -> {
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/modding/coremod/config/ControllerKeybindsConfigEntry.java:19]: Fix static import

File: src/main/java/legend/game/title/GameOver.java [hunk 24]
```diff
@@ -64,7 +64,7 @@ public static void gameOver() {
 
       case 3 -> {
         deallocateRenderables(0xff);
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/title/GameOver.java:67]: Fix static import

File: src/main/java/legend/game/title/Ttle.java [hunk 25]
```diff
@@ -283,7 +283,7 @@ public static void initializeMainMenu() {
       fireAnimation_800c66d4[i] = FUN_800cdaa0(rect, _800ce7b0[i]);
     }
 
-    scriptStartEffect(2, 15);
+    Scus94491BpeSegment.startFadeEffect(2, 15);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/title/Ttle.java:286]: Fix static import

File: src/main/java/legend/game/wmap/WMap.java [hunk 26]
```diff
@@ -494,7 +494,7 @@ public static void handleInventoryTransition() {
                         if(a0._05 == 0) {
                           if(mapState_800c6798._d8 == 0) {
                             if(a0._250 == 0) {
-                              scriptStartEffect(1, 15);
+                              Scus94491BpeSegment.startFadeEffect(1, 15);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/wmap/WMap.java:497]: Fix static import

File: src/main/java/legend/game/inventory/screens/StatusScreen.java [hunk 27]
```diff
@@ -42,7 +42,7 @@ public StatusScreen(final Runnable unload) {
   protected void render() {
     switch(this.loadingStage) {
       case 0 -> {
-        scriptStartEffect(2, 10);
+        Scus94491BpeSegment.startFadeEffect(2, 10);
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/inventory/screens/StatusScreen.java:45]: Fix static import

File: src/main/java/legend/game/combat/Bttl_800d.java [hunk 28]
```diff
@@ -2387,6 +2389,11 @@ public static void FUN_800d8bf4(final float x, final float y, final float z, fin
     cam.refpointCallbackIndex_121 = 23;
   }
 
+  @ScriptDescription("Causes the battle camera projection plane distance to begin moving")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "mode", description = "How the camera should move")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "newDistance", description = "The new projection plane distance")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "frames", description = "The number of frames it should take to change the distance")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "?")
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/combat/Bttl_800d.java:2396]: This is delta 1 in the step smoother

**Review by theflyingzamboni — CHANGES_REQUESTED** [review]
A lot of the comments are repetitive just to mark the locations so you can find all the instances.

File: src/main/java/legend/game/SMap.java [hunk 29]
```diff
@@ -1640,35 +1767,43 @@ public static FlowControl FUN_800df9a8(final RunningScript<?> script) {
...
-  public static FlowControl scriptMeShowModelPart(final RunningScript<?> script) {
+  public static FlowControl scriptSelfShowModelPart(final RunningScript<?> script) {
     script.params_20[1] = script.params_20[0];
     script.params_20[0] = new ScriptStorageParam(script.scriptState_04, 0);
     return scriptShowModelPart(script);
   }
 
   @Method(0x800dfbd4L)
-  public static FlowControl scriptMeFaceCamera(final RunningScript<?> script) {
+  public static FlowControl scriptSelfFaceCamera(final RunningScript<?> script) {
```
[the remark this answers is not in the document]
Comment (LordMonoxide) [reply src/main/java/legend/game/SMap.java:1807]: This one's done in a subsequent PR, I just happened to rename it here (doing it from the top of the subfuncs list down)

File: src/main/java/legend/game/SMap.java [hunk 30]
```diff
@@ -1478,80 +1556,104 @@ public static FlowControl FUN_800df488(final RunningScript<?> script) {
...
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "x", description = "The new camera X position")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "y", description = "The new camera Y position")
+  @ScriptParam(direction = ScriptParam.Direction.IN, type = ScriptParam.Type.INT, name = "z", description = "The new camera Z position")
   @Method(0x800df6a4L)
-  public static FlowControl FUN_800df6a4(final RunningScript<?> script) {
+  public static FlowControl scriptSetCameraPos(final RunningScript<?> script) {
     SetRotMatrix(worldToScreenMatrix_800c3548);
     SetTransMatrix(worldToScreenMatrix_800c3548);
-    FUN_800e8104(new SVECTOR().set((short)script.params_20[0].get(), (short)script.params_20[0].get(), (short)script.params_20[0].get()));
+    setCameraPos(new SVECTOR().set((short)script.params_20[0].get(), (short)script.params_20[0].get(), (short)script.params_20[0].get()));
```
[the remark this answers is not in the document]
Comment (LordMonoxide) [reply src/main/java/legend/game/SMap.java:1652]: This is retail, but it's gotta be a bug... we should look into it

**Pull request merged**